examples/p4est_2d_dgsem/elixir_euler_forward_step.jl

```diff
@@ -0,0 +1,150 @@
+
+using Downloads: download
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the compressible Euler
+
+equations = CompressibleEulerEquations2D(1.4)
+
+"""
+Compressible Euler setup for a Mach 3 wind tunnel flow with a forward facing step.
+Results in strong shock interactions as well as Kelvin-Helmholtz instabilities at later times.
+See Section IV b on the paper below for details.
+
+- Paul Woodward and Phillip Colella (1984)
+  The Numerical Simulation of Two-Dimensional Fluid Flows with Strong Shocks.
+  [DOI: 10.1016/0021-9991(84)90142-6](https://doi.org/10.1016/0021-9991(84)90142-6)
+"""
+@inline function initial_condition_mach3_flow(x, t, equations::CompressibleEulerEquations2D)
+  # set the freestream flow parameters
+  rho_freestream = 1.4
+  v1 = 3.0
+  v2 = 0.0
+  p_freestream = 1.0
+
+  prim = SVector(rho_freestream, v1, v2, p_freestream)
+  return prim2cons(prim, equations)
+end
+
+initial_condition = initial_condition_mach3_flow
+
+
+boundary_condition_inflow = BoundaryConditionDirichlet(initial_condition_mach3_flow)
+
+# Outflow boundary condition.
+# FIXME: For now only works for supersonic outflow where all values come from the internal state.
+# The bones are here for the subsonic outflow as well. One simply needs to pass the reference pressure
```

Comment:
What do you mean by "one simply needs to pass the reference pressure"? Would it, e.g., work by making this a callable struct, e.g.,
```julia

struct BoundaryConditionOutflow{T<:Real}
  p_reference::Real
end

function (bc::BoundaryConditionOutflow)(u_inner, normal_direction::AbstractVector, x, t,
                                        surface_flux_function, equations::CompressibleEulerEquations2D)
  # here use bc.p_reference
  # ...
end

my_p_reference = 1.234
boundary_condition_outflow = BoundaryConditionOutflow(my_p_reference)
```
?

Reply:
Here what I mean is that you must set the external pressure to be the  freestream pressure (called something like p_infinity) for the subsonic case. In FLUXO we have the ability to set such background flow state values that are then available for particular applications / BCs. We haven't needed such a functionality in Trixi up to this point so that is why I was unsure the best way to do this

Reply:
The subsonic outflow is a mixture of using internal solution values and the external background pressure whereas the supersonic outflow uses all internal values. So we could potentially use the formalism proposed but it is very compressible Euler specific setting the outflow in this way

Reply:
Going strictly by the YAGNI principle (you ain't gonna need it), we should only implement what we actually use. If you have no use for the subsonic case at the moment, I suggest to leave the code as it is right now and move on.

Reply:
Yeah, I don't have a good testcase off the top of my head that needs the subsonic outflow. We can leave the (untested) skeleton procedure where it is in the elixir

Reply:
Sounds good to me

Reply:
Yep (I'll close it)